### PR TITLE
streamaggr: made labels compressor shared

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Though labels compressor is quite resource intensive, each aggregator and deduplicator instance has it's own compressor. Made it shared across all aggregators to consume less resources while using multiple aggregators

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: app/vmagent/remotewrite/remotewrite.go [hunk 1]
```diff
@@ -121,6 +121,9 @@ var (
 	// rwctxsDefault contains statically populated entries when -remoteWrite.url is specified.
 	rwctxsDefault []*remoteWriteCtx
 
+	// labelsCompressor contains information about all compressed labels for streaming aggregation
+	labelsCompressor promutils.LabelsCompressor
```
Comment (hagen1778) [inline app/vmagent/remotewrite/remotewrite.go:125]: Can we make this as global var in `lib/streamaggr`, and define `vm_streamaggr_labels_compressor_size_bytes ` and `vm_streamaggr_labels_compressor_items_count ` there as well?

Comment (AndrewChubatiuk) [reply app/vmagent/remotewrite/remotewrite.go]: already made this changes, but just pushed them now

**Review by hagen1778 — APPROVED** [review]
LGTM!
only a changelog line is missing.

**Comment by AndrewChubatiuk** [comment]
> LGTM! only a changelog line is missing.

added

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,7 @@ See also [LTS releases](https://docs.victoriametrics.com/lts-releases/).
...
 * FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent/): support data replication additionally to sharding among remote storage systems if `-remoteWrite.shardByURLReplicas=N` command-line flag is set additionally to `-remoteWrite.shardByURL` command-line flag, where `N` is desired replication factor. This allows setting up data replication among failure domains when the replication factor is smaller than the number of failure domains. See [these docs](https://docs.victoriametrics.com/vmagent/#sharding-among-remote-storages) and [this feature request](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6054).
 * FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent/): reduce CPU usage when [sharding among remote storage systems](https://docs.victoriametrics.com/vmagent/#sharding-among-remote-storages) is enabled.
+* FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent/): reduce memory usage by making labels compressor for stream aggregators and deduplicators shared.
```
Comment (hagen1778) [inline docs/CHANGELOG.md:38]:
```suggestion
* FEATURE: [stream aggregation](https://docs.victoriametrics.com/stream-aggregation/): reduce memory usage during stream aggregation if multiple aggregation configs are used for the same set of data.
```

**Comment by hagen1778** [comment]
Thank you!

**Comment by valyala** [comment]
@AndrewChubatiuk , thanks for the excellent pull request, which reduces memory usage when multiple stream aggregations are configured!

**Pull request merged**